sambacc/smbconf_api.py

```diff
@@ -0,0 +1,73 @@
+#
+# sambacc: a samba container configuration tool
+# Copyright (C) 2023  John Mulligan
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>
+#
+
+import typing
+
+
+class ConfigStore(typing.Protocol):
+    def __getitem__(self, name: str) -> list[tuple[str, str]]:
+        ...  # pragma: no cover
+
+    def __setitem__(self, name: str, value: list[tuple[str, str]]) -> None:
+        ...  # pragma: no cover
+
+    def __iter__(self) -> typing.Iterator[str]:
+        ...  # pragma: no cover
+
+
+class SimpleConfigStore:
+    def __init__(self) -> None:
+        self._data: dict[str, list[tuple[str, str]]] = {}
+
+    @property
+    def writeable(self) -> bool:
+        """True if using a read-write backend."""
+        return True
+
+    def __getitem__(self, name: str) -> list[tuple[str, str]]:
+        return self._data[name]
+
+    def __setitem__(self, name: str, value: list[tuple[str, str]]) -> None:
+        self._data[name] = value
+
+    def __iter__(self) -> typing.Iterator[str]:
+        return iter(self._data.keys())
+
+    def import_smbconf(
+        self, src: ConfigStore, batch_size: typing.Optional[int] = None
+    ) -> None:
+        """Import content from one SMBConf configuration object into the
+        current SMBConf confiugration object.
```

Comment:
Trivial typo here: `confiugration` => `configuration`

HTH,
Matteo

Reply:
Thank you. I have fixed the mistake in PR #85 - this branch is based on the branch for PR #85 as this one was me experimenting with changes to the CI. I was being lazy when I filed it and did not cleanly separate the streams of work. When I next rebase the changes related to global config on AD DCs should disappear from this PR.